cmd/templ/generatecmd/main.go

```diff
@@ -348,8 +447,26 @@ func generate(ctx context.Context, basePath, fileName string, generateSourceMapV
 		return nil, fmt.Errorf("%s source formatting error: %w", fileName, err)
 	}
 
-	if err = os.WriteFile(targetFileName, data, 0644); err != nil {
-		return nil, fmt.Errorf("%s write file error: %w", targetFileName, err)
+	// Hash and write the file if the hash has changed
+	hash := md5.Sum(data)
+	if hashes[targetFileName] != hash {
+		if err = os.WriteFile(targetFileName, data, 0o644); err != nil {
+			return nil, fmt.Errorf("%s write file error: %w", targetFileName, err)
```

Comment:
All the other errors in templ are basically of the form `failed to ...`, so I'd prefer:

```suggestion
			return nil, fmt.Errorf("failed to write target file %q: %w", targetFileName, err)
```